auditlog/audit_log_record_repository.go

```diff
@@ -0,0 +1,136 @@
+package auditlog
+
+import (
+	"context"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/closeable"
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// RecordRepository provides functions to create and view Records
+type RecordRepository interface {
```

Comment:
Why are we defining the repository and the repository struct in two separate source files?

Reply:
I just wanted to keep the files smaller, but I'm also fine with keeping both the model/struct and the repository in the same file

Reply:
done in b78e640, renamed the `RecordRepository` to `Repository` since it's already in the `auditlog` pkg